lms/djangoapps/shoppingcart/tests/test_views.py

```diff
@@ -866,6 +871,101 @@ def test_show_receipt_json(self, num_items):
                 'line_desc': 'Honor Code Certificate for course Test Course'
             })
 
+    @ddt.data(0, 1, 2)
+    @override_settings(
+        ECOMMERCE_API_URL=EcommerceApiTestMixin.ECOMMERCE_API_URL,
+        ECOMMERCE_API_SIGNING_KEY=EcommerceApiTestMixin.ECOMMERCE_API_SIGNING_KEY
+    )
+    def test_show_ecom_receipt_json(self, num_items):
+        # set up the get request to return an order with X number of line items.
+
+        # Log in the student. Use a false order ID for the E-Commerce Application.
+        self.login_user()
+        url = reverse('shoppingcart.views.show_receipt', args=['EDX-100042'])
+        with self.mock_get_order(num_items=num_items):
+            resp = self.client.get(url, HTTP_ACCEPT="application/json")
+
+        # Should have gotten a successful response
+        self.assertEqual(resp.status_code, 200)
+
+        # Parse the response as JSON and check the contents
+        json_resp = json.loads(resp.content)
+        self.assertEqual(json_resp.get('currency'), self.mock_get_order.ORDER['currency'])
+        self.assertEqual(json_resp.get('total_cost'), self.mock_get_order.ORDER['total_excl_tax'])
+        self.assertEqual(json_resp.get('status'), self.mock_get_order.ORDER['status'])
+        self.assertEqual(json_resp.get('billed_to'), {
+            'first_name': self.mock_get_order.ORDER['billing_address']['first_name'],
+            'last_name': self.mock_get_order.ORDER['billing_address']['last_name'],
+            'street1': self.mock_get_order.ORDER['billing_address']['line1'],
+            'street2': self.mock_get_order.ORDER['billing_address']['line2'],
+            'city': self.mock_get_order.ORDER['billing_address']['line4'],
+            'state': self.mock_get_order.ORDER['billing_address']['state'],
+            'postal_code': self.mock_get_order.ORDER['billing_address']['postcode'],
+            'country': self.mock_get_order.ORDER['billing_address']['country']['display_name']
+        })
+
+        self.assertEqual(len(json_resp.get('items')), num_items)
+        for item in json_resp.get('items'):
+            self.assertEqual(item, {
+                'unit_cost': self.mock_get_order.LINE['unit_price_excl_tax'],
+                'quantity': self.mock_get_order.LINE['quantity'],
+                'line_cost': self.mock_get_order.LINE['line_price_excl_tax'],
+                'line_desc': self.mock_get_order.LINE['description']
+            })
+
+    class mock_get_order(object):  # pylint: disable=invalid-name
+        """Mocks calls to EcommerceAPI.get_order. """
+        patch = None
+
+        ORDER = copy.deepcopy(EcommerceApiTestMixin.ECOMMERCE_API_SUCCESSFUL_BODY)
+        ORDER['total_excl_tax'] = 40.0
+        ORDER['currency'] = 'USD'
+        ORDER['sources'] = [{'transactions': [{'date_created': datetime.now()}]}]
```

Comment:
this doesn't seem to be tested.  (might be better to define a specific datetime so equality assertions are easier?)